### ISSUE TITLE
Add indentation to procedure parameters

### ISSUE THREAD

**Issue by fdw** [sqlfluff/sqlfluff#3200]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

The parameter list for functions is indented, the list for procedures is not. For example:
```sql
create function good(
  @param1 int
) as
...

create procedure bad
@param1 int
as
```

Would it be possible to add the same indentation to procedure parameters

### Use case

First, I'd like to have a consistent style for parameters. Secondly, I find it easier to read if the parameters are indented.

### Dialect

I'm using T-SQL, but it should be applicable to all

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by fdw** [comment]
So, I take it you would accept this change 🙂

How would I go about adding it? Is it just adding `Indent` and `Dedent` around `Ref("ProcedureParameterListGrammar", optional=True),` in `CreateProcedureStatementSegment`?

**Comment by tunetheweb** [comment]
Yup. And a test case to `test/fixtures/rules/std_rule_cases/L003.yml`

**Comment by fdw** [comment]
Thank you, that works 🙂

Is there some way to force them to be on a new line (similar to `select`'s clauses)?

**Comment by tunetheweb** [comment]
No that’s a separate rule and currently only implemented for `SELECT`. There’s a few open issues asking for similar but no ones worked on them yet.

**Comment by fdw** [comment]
Ah, thanks. I had hoped this might work similar to `Indent` and `Dedent`. I'll open a PR for what I have now, this is enough for this issue 🙂
